DbAccessor/src/Messages/CreateConversation.cs

```diff
@@ -0,0 +1,119 @@
+using System.Text.Json;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using MongoDB.Driver;
+using MongoDB.Bson;
+using Microsoft.Azure.Functions.Worker;
+
+namespace GoRideShare
+{
+
+    public class CreateConversation
+    {
+        private readonly ILogger<CreateConversation> _logger;
+        private readonly string? _baseApiUrl;
+        // initialize the MongoDB client lazily. This is a best practice for serverless functions because it is not efficient to establish Mongo connections on every execution of our Azure Function
+        public static Lazy<MongoClient> lazyClient = new Lazy<MongoClient>(InitializeMongoClient);
+        public static MongoClient client = lazyClient.Value;
+
+        public static MongoClient InitializeMongoClient()
+        {
+            return new MongoClient(Environment.GetEnvironmentVariable("MONGODB_ATLAS_URI"));
+        }
+
+        public CreateConversation(ILogger<CreateConversation> logger)
+        {
+            _logger = logger;
+            _baseApiUrl = Environment.GetEnvironmentVariable("BASE_API_URL");
+        }
+        
+        [Function("CreateConversation")]
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "post")] HttpRequest req)
+        {
+            // If validation result is not null, return the bad request result
+            var validationResult = Utilities.ValidateHeaders(req.Headers, out string userId);
+            if (validationResult != null)
+            {
+                return validationResult;
+            }
+
+            // Read the request body to get the user's registration information
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+            ConversationRequest? convoRequest;
+            try
+            {
+                convoRequest = JsonSerializer.Deserialize<ConversationRequest>(requestBody);
+
+                var (invalid, errorMessage) = convoRequest.validate();
+                if (invalid)
+                {
+                    return new BadRequestObjectResult(errorMessage);
+                }
+            }
+            catch (JsonException ex)
+            {
+                _logger.LogError($"JSON deserialization failed: {ex.Message}");
+                return new BadRequestObjectResult("Incomplete Conversation Request data.");
+            }
+            _logger.LogInformation($"Raw Request Body: {JsonSerializer.Serialize(requestBody)}");
+
+
+            // Get the user name and photo details by calling SQL Get user endpoint
+            User otherUser;
+            string endpoint = $"{_baseApiUrl}/api/GetUser";
+
+            // make http request using req.Headers and endpoint to get the user details
+            var (error, response) = await Utilities.MakeHttpGetRequest(convoRequest.UserId, endpoint);
```

Comment:
This causes the UserId passed in the request object to be used for:
`request.Headers.Add("X-User-ID", xUserId);`
This doesn't seem right to me.